src/nvim/normal.c

```diff
@@ -4897,6 +4897,9 @@ static void nv_next(cmdarg_T *cap)
     (void)normal_search(cap, 0, NULL, SEARCH_MARK | cap->arg, NULL);
     cap->count1 -= 1;
   }
+  // Redraw the buffer to ensure the highlighting of the current match under
+  // the cursor as well as the old match both get properly updated
+  redraw_later(curwin, NOT_VALID);
```

Comment:
This a bit overzealous. instead use `redrawWinline` twice with the old and new lnum respectively.